src/esbuild.ts

```diff
@@ -39,49 +41,67 @@ async function readUserPlugins(): Promise<Partial<ReadUserPluginsReturnType>> {
   return defaultOut;
 }
 
-export async function buildESM(
-  srcFilesToCompile: string[],
-  outDir: string,
-  sourcemap: boolean,
-  target: string[],
-  platform: esbuild.Platform = 'browser',
-) {
+export interface BuildBaseArgs {
+  external: string[];
+  outDir: string;
+  platform?: esbuild.Platform;
+  sourcemap: boolean;
+  srcFilesToCompile: string[];
+  target: string[];
+}
+
+export async function buildESM({
+  external,
+  outDir,
+  sourcemap,
+  srcFilesToCompile,
+  target,
+  platform = DEFAULT_PLATFORM,
+}: BuildBaseArgs) {
   await esbuild.build({
-    platform,
-    sourcemap,
-    target,
     bundle: false,
     entryPoints: srcFilesToCompile,
+    external,
```

Comment:
This appears to cause an error, because external only makes sense with bundle:true apparently. So maybe only pass external in buildCJS?